metrics-exporter-dogstatsd/src/state.rs

```diff
@@ -96,7 +102,13 @@ impl State {
 
             active_counters += 1;
 
-            let result = writer.write_counter(&key, value, self.get_aggregation_timestamp());
+            let prefix = if key.name().starts_with("datadog.dogstatsd.client") {
+                &None
+            } else {
+                &self.config.global_prefix
+            }.as_deref();
```

Comment:
I think you can make this cleaner:

```suggestion
            let prefix = if key.name().starts_with("datadog.dogstatsd.client") {
                None
            } else {
                self.config.global_prefix.as_deref()
            };
```